msbuild/props/SixLabors.Global.props

```diff
@@ -0,0 +1,167 @@
+    <UseSharedCompilation>true</UseSharedCompilation>
+  </PropertyGroup>
+
+  <!-- Package references and additional files which are consumed by all projects. -->
+  <ItemGroup>
+    <PackageReference Update="Microsoft.NETFramework.ReferenceAssemblies" PrivateAssets="All" Version="1.0.0" IsImplicitlyDefined="true"  />
+    <PackageReference Include="StyleCop.Analyzers" PrivateAssets="All" Version="1.1.118" IsImplicitlyDefined="true" />
+    <AdditionalFiles Include="$(MSBuildThisFileDirectory)..\..\stylecop.json" />
+  </ItemGroup>
+
+  <!--Define OS platform conditions and constants.
+   https://docs.microsoft.com/en-us/visualstudio/msbuild/property-functions?view=vs-2019#msbuild-property-functions
+   https://docs.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.osplatform?view=net-5.0#properties
+   -->
+  <PropertyGroup>
+    <IsWindows Condition="$([MSBuild]::IsOSPlatform('Windows'))">true</IsWindows>
+    <IsOSX Condition="$([MSBuild]::IsOSPlatform('OSX'))">true</IsOSX>
+    <IsLinux Condition="$([MSBuild]::IsOSPlatform('Linux'))">true</IsLinux>
+    <IsFreeBSD Condition="$([MSBuild]::IsOSPlatform('FreeBSD'))">true</IsFreeBSD>
+  </PropertyGroup>
+  <Choose>
+    <When Condition="'$(IsWindows)'=='true'">
+      <PropertyGroup>
+        <DefineConstants>$(DefineConstants);OS_WINDOWS</DefineConstants>
+      </PropertyGroup>
+    </When>
+    <When Condition="'$(IsOSX)'=='true'">
+      <PropertyGroup>
+        <DefineConstants>$(DefineConstants);OS_OSX</DefineConstants>
+      </PropertyGroup>
+    </When>
+    <When Condition="'$(IsLinux)'=='true'">
+      <PropertyGroup>
+        <DefineConstants>$(DefineConstants);OS_LINUX</DefineConstants>
+      </PropertyGroup>
+    </When>
+    <When Condition="'$(IsFreeBSD)'=='true'">
+      <PropertyGroup>
+        <DefineConstants>$(DefineConstants);OS_FREEBSD</DefineConstants>
+      </PropertyGroup>
+    </When>
+  </Choose>
+
+  <!-- Define target framework specific constants.
+    https://apisof.net/
+    +===================+=======+==========+=====================+=============+=================+====================+==============+=========+============|
+    | SUPPORTS          | MATHF | HASHCODE | EXTENDED_INTRINSICS | SPAN_STREAM | ENCODING_STRING | RUNTIME_INTRINSICS | CODECOVERAGE | HOTPATH | CREATESPAN |
+    +===================+=======+==========+=====================+=============+=================+====================+==============+=========|============|
+    | netcoreapp3.1     |   Y   |    Y     |         Y           |      Y      |        Y        |        Y           |      Y       |    Y    |      Y     |
```

Comment:
What will be the pricess of maintaining this list efficiently? Can we just extend the repo-specific `.props` in individual PRs first, 
 without being blocked on the SharedInfrastructure stuff? (And migrate the options in follow up PRs)

Reply:
You can extend this with your own ones in repo-specific `.props` since we use `$(DefineConstants)`. I'd only ever update this if we find we are reusing several identical preprocessor constants across repos again.